demo/openssl-qat-engine/Dockerfile

```diff
@@ -1,4 +1,65 @@
-FROM ubuntu:24.04
+FROM ubuntu:24.04 AS builder
+
+ARG QATLIB_VERSION="24.09.0"
+ARG QAT_ENGINE_VERSION="v1.6.2"
+ARG IPSEC_MB_VERSION="v1.5"
+ARG IPP_CRYPTO_VERSION="cryptography-primitives_1_0_0"
 
 RUN apt update && \
-    apt install --no-install-recommends -y qatengine qatlib-examples qatzip openssl
+    env DEBIAN_FRONTEND=noninteractive apt install -y \
+    git \
+    autoconf \
+    libtool \
+    libssl-dev \
+    libnuma-dev \
+    nasm \
+    pkg-config \
+    make \
+    zlib1g-dev \
+    cmake \
+    python3
+
+RUN git clone --depth 1 -b $QAT_ENGINE_VERSION https://github.com/intel/QAT_Engine && \
+    git clone --depth 1 -b $IPP_CRYPTO_VERSION https://github.com/intel/ipp-crypto && \
+    git clone --depth 1 -b $IPSEC_MB_VERSION https://github.com/intel/intel-ipsec-mb && \
+    git clone --depth 1 -b $QATLIB_VERSION https://github.com/intel/qatlib
+
+RUN cd /qatlib && \
+    ./autogen.sh && \
+    ./configure \
+    --prefix=/usr \
+    --enable-systemd=no && \
+    make -j && \
+    make install samples-install
+
+RUN cd /ipp-crypto/sources/ippcp/crypto_mb && \
+    cmake . -B"../build" \
+    -DOPENSSL_INCLUDE_DIR=/usr/include/openssl \
+    -DOPENSSL_LIBRARIES=/usr/lib64 \
+    -DOPENSSL_ROOT_DIR=/usr/bin/openssl && \
+    cd ../build && \
+    make crypto_mb && make install
+
+RUN cd /intel-ipsec-mb && \
+    make && make install LIB_INSTALL_DIR=/usr/lib64
+
+RUN cd /QAT_Engine && \
+    sed -i '/fprintf(stderr, "QAT_HW device not available & QAT_SW not enabled. Using OpenSSL_SW!\\n");/ s/$/ return ret;/' e_qat.c && \
+    ./autogen.sh && \
+    ./configure \
+    --disable-qat_sw && \
```

Comment:
let's build with a custom `--with-qat_engine_id=` name. Then for the final image, you can install the default `qatengine` and the custom one built here.